### PR TITLE
cfx-ui: implement a close shortcut for lazy people

### PR DESCRIPTION
**Pull request by PichotM**
For lazy people

### CODE REVIEW

File: ext/cfx-ui/src/app/servers/connecting-popup.component.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ export class ConnectingPopupComponent implements OnInit {
 	closeLabel = "#Servers_CloseOverlay";
 	retryLabel = "#Servers_Retry";
 	submitting = false;
+	closeKeys = [27, 8, 46];
```
Comment (blattersturm) [inline ext/cfx-ui/src/app/servers/connecting-popup.component.ts:22]: Magic numbers - everyone's favorite! `event.key` or `event.code` might help with readability as it'd not involve uncommented numbers?

File: ext/cfx-ui/src/app/servers/connecting-popup.component.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<div [class.overlay]="!minMode" [class.minmode]="minMode" [class.active]="showOverlay">
+<div [class.overlay]="!minMode" [class.minmode]="minMode" [class.active]="showOverlay" (keydown)="onKeyPress($event)" tabindex="1">
```
Comment (blattersturm) [inline ext/cfx-ui/src/app/servers/connecting-popup.component.html:1]: Does `tabindex` in this case make it automatically take focus when added to the DOM?

**Comment by PichotM** [comment]
Done

File: ext/cfx-ui/src/app/servers/connecting-popup.component.ts [hunk 3]
```diff
@@ -314,4 +315,10 @@ export class ConnectingPopupComponent implements OnInit {
 	doRetry() {
 		this.retry.emit();
 	}
+
+    onKeyPress(event: KeyboardEvent) {
+		if (this.closeKeys.includes(event.code)) {
```
Comment (thers) [inline ext/cfx-ui/src/app/servers/connecting-popup.component.ts:320]: You shouldn't be using `event.code` as it ignores user layout and instead maps to the physical layout. Use `event.key` instead.

File: ext/cfx-ui/src/app/servers/connecting-popup.component.ts [hunk 4]
```diff
@@ -19,6 +19,7 @@ export class ConnectingPopupComponent implements OnInit {
 	closeLabel = "#Servers_CloseOverlay";
 	retryLabel = "#Servers_Retry";
 	submitting = false;
+	closeKeys = ["Escape", "Backspace", "Delete"];
```
Comment (thers) [inline ext/cfx-ui/src/app/servers/connecting-popup.component.ts:22]: Err, backspace and delete for closing modal? That's something unusual.

**Pull request merged**